packages/core/src/modules/sd-jwt-vc/credential-status/token-status-list/TokenStatusList.ts

```diff
@@ -0,0 +1,237 @@
+    // update indices
+    for (const revokedIndex of options.indices) {
+      statusList.setStatus(revokedIndex, 1)
+    }
+
+    const issuerKey = await sdJwtService.extractKeyFromIssuer(agentContext, issuer, true)
+
+    // construct jwt payload
+    const jwtPayload = new JwtPayload({
+      iss,
+      iat: Math.floor(Date.now() / 1000),
+      additionalClaims: {
+        status_list: {
+          bits: statusList.getBitsPerStatus(),
+          lst: statusList.compressStatusList(),
+        },
+      },
+    } satisfies TokenStatusListJwtPayload)
+
+    const jwt = await jwsService.createJwsCompact(agentContext, {
+      payload: jwtPayload,
+      keyId: issuerKey.publicJwk.keyId,
+      protectedHeaderOptions: {
+        alg: issuerKey.alg,
+        typ: header.typ,
+        kid: issuerKey.kid,
+      },
+    })
+
+    if (options.publish) {
+      await this.publishStatusList(agentContext, issuer, jwt, publishOptions)
+    }
+
+    return true
+  }
+
+  async getStatus(agentContext: AgentContext, uri: string, index: number): Promise<number> {
+    const currentStatusListJwt = await this.getStatusList(agentContext, uri)
+    const statusList = getListFromStatusListJWT(currentStatusListJwt)
+    return statusList.getStatus(index)
+  }
+
+  async getStatusList(agentContext: AgentContext, uri: string): Promise<string> {
+    const registry = this.findRegistry({ uri })
+    if (!registry) {
+      throw new TokenStatusListError(`No token status list registry registered for uri ${uri}`)
+    }
+
+    return await registry.retrieve(agentContext, uri)
+  }
+
+  getStatusListFetcher(agentContext: AgentContext) {
+    return async (uri: string) => {
+      return await this.getStatusList(agentContext, uri)
+    }
+  }
+
+  async publishStatusList(
+    agentContext: AgentContext,
+    issuer: SdJwtVcIssuer,
+    jwt: string,
+    options: PublishTokenStatusListOptions
+  ): Promise<string> {
+    // verify jwt
+    const verified = await this.verifyStatusList(agentContext, issuer, jwt)
+    if (!verified.isValid) {
+      throw new TokenStatusListError('Invalid Jwt in the provided statusListUri')
+    }
+
+    const registry = this.findRegistry({ uri: options.uri, issuer })
+    if (!registry) {
+      throw new TokenStatusListError(`No token status list registry registered for issuer ${issuer}`)
+    }
+    return await registry.publish(agentContext, issuer, jwt, options)
+  }
+
+  private findRegistry({ uri, issuer }: { uri?: string; issuer?: SdJwtVcIssuer }): TokenStatusListRegistry | null {
+    let method: string
+
+    if (uri) {
+      if (isDid(uri)) {
+        method = parseDid(uri).method
+      } else if (isURL(uri)) {
+        method = 'http'
+      } else {
+        throw new TokenStatusListError('Status List Uri is not supported')
+      }
+    } else if (issuer && issuer.method === 'did') {
+      method = parseDid(issuer.didUrl).method
+    } else {
+      throw new TokenStatusListError('Status List Uri is not provided')
+    }
```

Comment:
Adding this logic here, makes the registry not really extendable by nature. 

I think it would make more sense to provide an uri to the registry and match based on regex (like we do with anoncreds). Both the `iss`/`uri` values can be passed to it.

Reply:
@TimoGlastra Made the changes